### ISSUE TITLE
Backspace key not working

### ISSUE THREAD

**Issue by chamburr** [Papierkorb/fancyline#9]
Under [`src/fancyline/key.cr`](https://github.com/Papierkorb/fancyline/blob/master/src/fancyline/key.cr), backspace is defined with the keycode 127, which seems to result in the correct behaviour for Windows and macOS terminals. However, the keycode 8 is sent on many Linux systems instead, and it is not being handled correctly.

Example on Kali Linux for backspace:
<img width="445" alt="Screenshot 2021-04-03 at 10 28 41 AM@2x" src="https://user-images.githubusercontent.com/42373024/113465541-627f8a00-9467-11eb-8035-76dd1ef7f023.png">

Note: The [custom key binding example](https://github.com/Papierkorb/fancyline#step-3-custom-key-bindings) should probably be changed to avoid confusion when this is fixed.

**Comment by lagerfeuer** [comment]
Hey @chamburr, sorry for the late response, don't have much time outside of work. I'll look into it when I get a chance. Which version of Crystal are you using? I haven't tested this lib with the latest release yet.

**Comment by chamburr** [comment]
Hi, thanks for the response. I'm using Crystal 1.0.

**Comment by lagerfeuer** [comment]
I figured, thanks for the response. Will look into it ASAP.

**Comment by lagerfeuer** [comment]
@chamburr I updated the library (finally), let me know if you still run into issues! Closing in the meantime, feel free to reopen the issue though.

**Comment by chamburr** [comment]
@lagerfeuer Hi, I tested the libary on the latest version and the issue is not fixed. I see that the only change was to explicitly allow Crystal 1.0, but that's likely not the cause of this problem.

I think the changes that need to be made are:
- Alias key code 8 to 127
- Disable Ctrl-H since that's backspace
- Update the key binding example

I'd love to help but I'm not really confident in Crystal. Also, I can't reopen the issue.

**Comment by lagerfeuer** [comment]
@chamburr Oh I see, I misunderstood what was going on. I reopened it. The code is not too hard to read, but if you don't feel comfortable, I'll look into it when I get a chance. Thanks for reporting and re-confirming!

**Comment by lagerfeuer** [comment]
I actually did look into it just now because it bugged me. It's working fine for me under Arch Linux. 
![image](https://user-images.githubusercontent.com/5903157/122856908-a2774000-d2e5-11eb-952e-9d2fab93a377.png)

Could you show me the output of `stty -a`? And also system information please, including which shell you are using.
I find it quite odd that it seems somewhat arbitrary whether it's `^H` or `^?`.
[This post](https://unix.stackexchange.com/questions/303016/backspace-not-working-in-kali-linux-terminal-hosted-on-backbox-using-virtual-box) seems to explain it very well. I don't know why it is happening, but it seems to be up to the OS to decide. The post also shows how to change it.

Bottom line, this seems like a system issue rather than a Crystal or fancyline issue, but I'd like to see this through in case someone else has the same problem.

Quick edit: We might be able to fix it regardless of system by finding a way to read the correct keycode from the system and using that instead of the hardcoded value.

**Comment by chamburr** [comment]
The bug was originally reported from a user on Arch Linux. I happened to have a Kali image so I installed it in a VM to test. Since it's reproducible, I believe the ambiguity is intended and needs to be handled by the library. I don't have the VM anymore and will try to get the information from the user.

Meanwhile, the following posts should explain this better:
https://stackoverflow.com/questions/4363309/how-to-check-for-the-backspace-character-in-c#comment4749488_4363345
https://stackoverflow.com/questions/44943249/detecting-key-backspace-in-ncurses/44943460#44943460

**Comment by chamburr** [comment]
Here's an update from the user and it appears to me that `erase = ^H;` is what you're looking for.

![](https://chamburr.xyz/u/mBgixr.png)

**Comment by lagerfeuer** [comment]
Actually, I think that's plenty of information. My proposed solution would be to simply add `^H` as backspace. I use both backspace and <kbd>Ctrl</kbd> + <kbd>H</kbd> a lot every day, even though they send different keycodes on my system (which I never thought about until you raised this issue). This way we would cover both stty settings and still allow the manual use of <kbd>Ctrl</kbd> + <kbd>H</kbd>.

**Comment by lagerfeuer** [comment]
@chamburr I opened PR #11 to deal with this, can you confirm that it works on your system as well? The fix is in the branch `backspace-issue`.

**Comment by chamburr** [comment]
I can confirm it works. Thanks!
